### PR TITLE
[fix] Apply all fixits

### PR DESCRIPTION
**Pull request by bruntib**
Clang tidy may provide automatically applicable fixits. These can be applied by clang-apply-replacements tool. This tool gets a directory as input which contains .yaml files describing the replacements. If any of these .yaml files can't apply for some reason, then the rest is skipped.

This clang tool is added a flag named --ignore-insert-conflict which ignores the error event and continues the application of further .yaml files. If CodeChecker recognizes the existence of this flag then it will be used, otherwise CodeChecker creates a temporary directory with a single .yaml file and applies them one by one.

### CODE REVIEW

**Review by Szelethus — CHANGES_REQUESTED** [review]
Well... do you have tests for this? :)

File: analyzer/codechecker_analyzer/cmd/fixit.py [hunk 1]
```diff
@@ -347,6 +377,13 @@ def main(args):
         LOG.error("JSON format error on standard input: %s", ex)
         sys.exit(1)
 
+    # "CodeChecker fixit" can be applied on the output of
+    # "CodeChecker cmd diff" command. Earlier this was a simple list of
+    # reports, but now this is an object with the list of reports and a version
+    # number indicating the version of the object's format.
+    if reports is not None and 'version' in reports:
```
Comment (Szelethus) [inline analyzer/codechecker_analyzer/cmd/fixit.py:384]:
```suggestion
    if reports and 'version' in reports:
```

File: analyzer/tests/functional/fixit/test_fixit.py [hunk 2]
```diff
@@ -328,3 +329,81 @@ def test_fixit_by_diff(self):
...
+            encoding="utf-8",
+            errors="ignore")
+        process.communicate()
+
+        # --- Test fixit --- #
+
+        orig_hash_1 = content_hash(source_file1)
+        orig_hash_2 = content_hash(source_file2)
+
+        # Toudh file so "CodeChecker fixit" doesn't apply on the first file.
```
Comment (Szelethus) [inline analyzer/tests/functional/fixit/test_fixit.py:388]:
```suggestion
        # Touch file so "CodeChecker fixit" doesn't apply on the first file.
```

**Review by Szelethus — APPROVED** [review]
LGTM

**Review by vodorok — APPROVED** [review]
LGTM

**Pull request merged**